cli/command/stack/kubernetes/conversion.go

```diff
@@ -65,13 +68,43 @@ func toSwarmProtocol(protocol apiv1.Protocol) swarm.PortConfigProtocol {
 	return swarm.PortConfigProtocol("unknown")
 }
 
-func fetchPods(namespace string, pods corev1.PodInterface) ([]apiv1.Pod, error) {
-	labelSelector := labels.SelectorForStack(namespace)
-	podsList, err := pods.List(metav1.ListOptions{LabelSelector: labelSelector})
+func fetchPods(stackName string, pods corev1.PodInterface, f filters.Args) ([]apiv1.Pod, error) {
+	services := f.Get("service")
+	// for existing script compatibility, support either <servicename> or <stackname>_<servicename> format
+	stackNamePrefix := stackName + "_"
+	for _, s := range services {
+		if strings.HasPrefix(s, stackNamePrefix) {
+			services = append(services, strings.TrimPrefix(s, stackNamePrefix))
+		}
+	}
+	listOpts := metav1.ListOptions{LabelSelector: labels.SelectorForStack(stackName, services...)}
+	var result []apiv1.Pod
+	podsList, err := pods.List(listOpts)
 	if err != nil {
 		return nil, err
 	}
-	return podsList.Items, nil
+	nodes := f.Get("node")
+	for _, pod := range podsList.Items {
+		if filterPod(pod, nodes) &&
+			// name filter is done client side for matching partials
+			f.FuzzyMatch("name", stackNamePrefix+pod.Name) {
```

Comment:
So does this mean that we require the name field to be like `--filter name=<stack name prefix> + <pod name>`?

Reply:
This is a `FuzzyMatch` meaning with a `stackNamePrefix` of `dtcccccc` and a `pod.Name` of `backend-695899db65-tptbg` all these `--filter` will match:
- `name=dtc`
- `name=dtcccccc`
- `name=dtcccccc-backend`
- `name=dtcccccc-backend-695899db65-tptbg`

However this one won't:
- `name=backend`

This seems to be consistent with the behavior with `--orchestrator=swarm`, however @simonferquel  initially put `Match` here thus allowing the following filters to match:
- `name=backend`
- `name=ccc`

etc…

I'm not sure which solution we want, the specification document was a bit vague and gave no example of expected outputs:
> * Best effort mapping of --filter flag
>   * Best effort as some filters will not map perfectly

Reply:
Wait, aren't you just concatenating the `stackNamePrefix` and `pod.Name` with `stackNamePrefix+pod.Name`? You said that `dtcccccc-backend` should match, implying there should be a hyphen between `stackNamePrefix` and `pod.Name`, but that doesn't seem to be the case.

Reply:
`stackNamePrefix` actually ends with an hyphen, it's defined a few lines before as
```
stackNamePrefix := stackName + "_"
```

Reply:
Not sure if this is intentional, but that's an underscore, not a hyphen.

Reply:
Sorry for the confusion, I made an error when writing the examples in https://github.com/docker/cli/pull/1023#discussion_r187767730: the hyphens should have been underscores i.e.
```
with a `stackNamePrefix` of `dtcccccc` and a `pod.Name` of `backend-695899db65-tptbg` all these `--filter` will match:
- `name=dtc`
- `name=dtcccccc`
- `name=dtcccccc_backend`
- `name=dtcccccc_backend-695899db65-tptbg`
```